tests/testsuite/artifact_dir.rs

```diff
@@ -304,6 +304,55 @@ For more information, try '--help'.
         .run();
 }
 
+#[cargo_test]
+fn deprecated_out_dir() {
```

Comment:
nit: not important enough for this change but in the future, I generally recommend adding the tests as  the first commit (with them passing).  Then as you change behavior, that is reflected in how the tests are changed to ensure they still pass.  This makes it easier for reviewers to see how behavior changed, we can then show those diffs in posts to the community for them to understand changes, and it helps "test the test" (I've made changes where I thought I was testing the right condition and wasn't).